classes/table/notifications/notifications_display.php

```diff
@@ -0,0 +1,58 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Renderable to render the list of observation time slot notifications
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace mod_observation\table\notifications;
+
+defined('MOODLE_INTERNAL') || die;
+
+/**
+ * Functions to view observation time slot notifications
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class notifications_display {
+
+    /**
+     * Creates a table that displays all the observation notifications for a users timeslot registration
+     * @param int $observationid ID of the observation instance to get the observation time slots and notifications from.
+     * @param int $userid ID of the user to find notifications for
+     * @param \moodle_url $callbackurl URL for action buttons in table to callback to
+     */
+    public static function notifications_table(int $observationid, int $userid, \moodle_url $callbackurl) {
```

Comment:
This feels like it should be in a renderer class, either renderer.php or classes/output/renderer.php

https://docs.moodle.org/dev/Renderer